### PR TITLE
Call reset instead of clear on signout

### LINKED ISSUE #44: call store.reset({name: hoodie.account.id}) instead of hoodie.store.clear() on signout
follow up for https://github.com/hoodiehq/hoodie-client/issues/43

We currently call `hoodie.store.clear()` on `signout` in [index.js](https://github.com/hoodiehq/hoodie-client/blob/master/index.js). `hoodie.store.clear` no longer exists, it was replaced with `.reset(options)` so we can set a new db name after signout, as `hoodie.account.id` changes as well.
 
---

- [x] claim this issue (@nickcolley)
- [x] update `"hoodie-client-store"` version in [package.json](https://github.com/hoodiehq/hoodie-client/blob/master/package.json) to the latest (`4.0.0` right now) unless it’s already updated
- [x] commit the change with `chore(package): hoodie-client-store@^4.0.0` (or whatever is the current version)
- [x] start a PR (#52)
- [x] replace the `hoodie.store.clear()` call with the `.reset()` method and pass in the new hoodie.account.id
- [x] commit the change with `fix: reset store on signout`
- [ ] review & merge
- [ ] update https://github.com/gr2m/milestones/issues/88

If you have any questions, ping us at http://hood.ie/chat/ or https://twitter.com/hoodiehq

### PR DESCRIPTION
**Pull request by nickcolley**
closes #44 

Out of curiosity why does hoodie.account.id change on signout?

### CODE REVIEW

**Comment by nickcolley** [comment]
Needs a review on the tests, the mocking seems a little suspect to me.

**Comment by HipsterBrown** [comment]
So this has tests to check that `store.reset` is called during `signIn` and `signOut`, correct? I'm also not following the `hoodie` mock passed into `init`.

**Comment by gr2m** [comment]
> Out of curiosity why does hoodie.account.id change on signout?

`hoodie.account.id` gets set immediately on first page load, even before you sign up for an account. So when you sign out, it’s as if you’d load hoodie for the first time: you get a new user account id. The main reason for this is that we want to allow users to create data even before signing up for an account, and store their user id in `createdBy` properties of all objects stored

File: lib/init.js [hunk 1]
```diff
@@ -8,7 +8,9 @@ function init (hoodie) {
   account.on('signout', function () {
     // TODO: prevent data loss on sign out:
     //       https://github.com/hoodiehq/hoodie-client/issues/22
-    store.clear()
+    store.reset({
+      name: hoodie.account.id
+    })
```
Comment (gr2m) [inline lib/init.js]: we need to pass in hoodie.account.id on `signin` as well, as the `hoodie.account.id` changes to the one of the account you sign into

Comment (nickcolley) [inline lib/init.js]: Thanks for the clarification of `hoodie.account.id` makes things clearer now!

I'll update the patch with your feedback.

**Comment by gr2m** [comment]
we mock the `hoodie` api for the `init` unit test to avoid any side effects, as it’s just a unit test, and ideally you pass in only the bits the respective unit test needs.

`hoodie-client` is kinda hard to test, and we still figure out how to do the implementation best, but for now the `init` test is better than nothing :)

**Comment by HipsterBrown** [comment]
Looks good to me. :+1:

**Comment by gr2m** [comment]
looks great, thanks @nickcolley

**Pull request merged**